From 2cec6525e7cd91686ce8561f8a8b9262dcb1dc1f Mon Sep 17 00:00:00 2001
From: tstegema <tstegema@90b65887-3827-0410-9a23-83215b262276>
Date: Tue, 25 Sep 2012 15:34:41 +0000
Subject: [PATCH 0350/1077] Mantis 0016225: MAT_CLA not visible in appcontrol
 fix openorb openorb sources are not compatible to java 1.5 => set
 source="1.4" in build.xml check for NullPointer in MergeStack

git-svn-id: svn://localhost/SelfServiceCommon/trunk@416 90b65887-3827-0410-9a23-83215b262276
---
 .../java/OpenORB-1.3.1/OpenORB/src/build.xml   | 18 ++++++++++++------
 .../src/main/org/openorb/util/MergeStack.java  |  5 ++++-
 2 files changed, 16 insertions(+), 7 deletions(-)

diff --git a/SelfServiceCommon/Massai/java/OpenORB-1.3.1/OpenORB/src/build.xml b/SelfServiceCommon/Massai/java/OpenORB-1.3.1/OpenORB/src/build.xml
index f0fe0d22..8964b456 100644
--- a/SelfServiceCommon/Massai/java/OpenORB-1.3.1/OpenORB/src/build.xml
+++ b/SelfServiceCommon/Massai/java/OpenORB-1.3.1/OpenORB/src/build.xml
@@ -407,7 +407,8 @@ public class ReleaseInfo
            destdir="${compiler-override-classes}"
            debug="${build.debug}"
            deprecation="${build.deprecation}"
-           optimize="${build.optimize}" >
+           optimize="${build.optimize}" 
+           source="1.4" >
       <classpath refid="project.class.path" />
     </javac>
 
@@ -420,7 +421,8 @@ public class ReleaseInfo
            debug="${build.debug}"
            deprecation="${build.deprecation}"
            optimize="${build.optimize}"
-           bootclasspathref="bootstrap-classpath" >
+           bootclasspathref="bootstrap-classpath" 
+           source="1.4" >
       <include name="org/openorb/ReleaseInfo.java" />
       <classpath refid="project.class.path" />
     </javac>
@@ -489,7 +491,8 @@ public class ReleaseInfo
            debug="${build.debug}"
            deprecation="${build.deprecation}"
            optimize="${build.optimize}"
-           bootclasspathref="bootstrap-classpath" >
+           bootclasspathref="bootstrap-classpath"
+           source="1.4" >
       <include name="org/openorb/compiler/" />
       <exclude name="org/openorb/compiler/doc/" />
       <exclude name="org/openorb/compiler/ir/" />
@@ -772,7 +775,8 @@ done
            debug="${build.debug}"
            deprecation="${build.deprecation}"
            optimize="${build.optimize}"
-           bootclasspathref="bootstrap-classpath" >
+           bootclasspathref="bootstrap-classpath"
+           source="1.4" >
       <classpath refid="project.class.path" />
     </javac>
   </target>
@@ -880,7 +884,8 @@ done
            debug="${build.debug}"
            deprecation="${build.deprecation}"
            optimize="${build.optimize}"
-           bootclasspathref="bootstrap-classpath" >
+           bootclasspathref="bootstrap-classpath"
+           source="1.4" >
       <classpath refid="project.class.path" />
     </javac>
   </target>
@@ -929,7 +934,8 @@ done
            debug="${build.debug}"
            deprecation="${build.deprecation}"
            optimize="${build.optimize}"
-           bootclasspathref="bootstrap-classpath" >
+           bootclasspathref="bootstrap-classpath"
+           source="1.4" >
       <classpath refid="project.class.path" />
     </javac>
   </target>
diff --git a/SelfServiceCommon/Massai/java/OpenORB-1.3.1/OpenORB/src/main/org/openorb/util/MergeStack.java b/SelfServiceCommon/Massai/java/OpenORB-1.3.1/OpenORB/src/main/org/openorb/util/MergeStack.java
index 9273c159..854b6fe1 100644
--- a/SelfServiceCommon/Massai/java/OpenORB-1.3.1/OpenORB/src/main/org/openorb/util/MergeStack.java
+++ b/SelfServiceCommon/Massai/java/OpenORB-1.3.1/OpenORB/src/main/org/openorb/util/MergeStack.java
@@ -435,7 +435,10 @@ public class MergeStack
             else
                 m_next = next;
 
-            m_next.m_prev = this;
+            if ( m_next != null )
+            {
+                m_next.m_prev = this;
+            }
         }
     }
 
-- 
2.41.0.windows.1

